### PR TITLE
refactor tensorflow to build from sources (and c interface)

### PR DESCRIPTION
**Pull request by davidlange6**
belatedly following the recipe of @mharrend 

@smuzaffar - I don't think the java-env works properly yet ..

### CODE REVIEW

**Comment by davidlange6** [comment]
nothing here is yet brought into the cmssw tool set - so no point in the tests to run

**Comment by davidlange6** [comment]
please test

**Comment by davidlange6** [comment]
i don't see why the tests should have run tensorflow-c at all.. (I haven't yet tested anything on slc6)

**Comment by smuzaffar** [comment]
Its becase of
```py2-pippkgs_depscipy.spec:BuildRequires: py2-tensorflow```
we need to add this dependency for slc7 only

**Comment by davidlange6** [comment]
Weird..But I didn't change that. Will look again....

Cheers,
David


On 30 Jun 2017, at 16:49, Malik Shahzad Muzaffar <notifications@github.com<mailto:notifications@github.com>> wrote:


Its becase of
py2-pippkgs_depscipy.spec:BuildRequires: py2-tensorflow
we need to add this dependency for slc7 only

**Comment by davidlange6** [comment]
pippkgs_depscipy is ok

**Comment by davidlange6** [comment]
whoops - and continuing my message "and indeed the logs don't show py2-tensorflow being built. Must be a typo someplace - will reproduce

File: py2-tensorflow.spec [hunk 1]
```diff
@@ -1,13 +1,25 @@
...
+### RPM external py2-tensorflow 1.1.0
 ## INITENV +PATH PYTHONPATH %{i}/${PYTHON_LIB_SITE_PACKAGES}
 
-# nothing on pypi is ok for tensorflow at this point
-%define tensorflowSource https://storage.googleapis.com/tensorflow/linux/cpu/tensorflow-1.0.0-cp27-none-linux_x86_64.whl
-%define PipDownloadOptions ALTSRC%%20%tensorflowSource%%20--no-deps%%20--no-binary%%3D:none:
-%define pip_name tensorflow
+Source: none
+
+Requires: tensorflow-sources
```
Comment (davidlange6) [inline py2-tensorflow.spec]: and this should be BuildRequires

**Comment by davidlange6** [comment]
ok - found my typo.. sorry for the noise

**Comment by davidlange6** [comment]
please test

File: java-env.spec [hunk 2]
```diff
@@ -1,22 +1,21 @@
 ### RPM external java-env 1.0
-## INITENV SET JAVA_HOME /usr/lib/jvm/java-1.8.0-openjdk-1.8.0.131-2.b11.el7_3.x86_64
+## INITENV SET JAVA_HOME %{java_home}
+## INITENV +PATH PATH %{java_home}/bin
+## INITENV +PATH LD_LIBRARY_PATH %{java_home}/lib
+## NOCOMPILER
 
+%define java_home /usr/lib/jvm/java
```
Comment (davidlange6) [inline java-env.spec]: this is not sufficient. needs to be jdk. We agreed to use the openjdk 1.8 before.

Comment (smuzaffar) [reply java-env.spec]: yes we need jdk-devel package and if that is missing then %{java_home}/bin/javac should fail

**Comment by davidlange6** [comment]
please test
(I think everything is now working on slc6 (disabled build) and 7. I'll start a new PR for enabling slc6 once this one goes ok.

**Comment by davidlange6** [comment]
ok - indeed it works - i guess we don't care about java/openjdk in the end.. Thanks

**Comment by davidlange6** [comment]
merge




On 1 Jul 2017, at 00:25, cmsbuild <notifications@github.com<mailto:notifications@github.com>> wrote:


Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3136/21041/summary.html

Comparison Summary:

  *   No significant changes to the logs found
  *   Reco comparison results: 0 differences found in the comparisons
  *   DQMHistoTests: Total files compared: 22
  *   DQMHistoTests: Total histograms compared: 1756063
  *   DQMHistoTests: Total failures: 15020
  *   DQMHistoTests: Total nulls: 0
  *   DQMHistoTests: Total successes: 1740877
  *   DQMHistoTests: Total skipped: 166
  *   DQMHistoTests: Total Missing objects: 0
  *   Checked 90 log files, 14 edm output root files, 22 DQM output files

File: tensorflow-sources.spec [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+%prep
+
+%setup -q -n tensorflow-%{realversion}
+
+%build
+export PYTHON_BIN_PATH=`which python`
+export TF_NEED_JEMALLOC=0
+export TF_NEED_HDFS=0
+export CC_OPT_FLAGS="-march=native"
```
Comment (slava77) [inline tensorflow-sources.spec:15]: I guess this is connected to the "illegal instruction" crash I get in https://github.com/cms-sw/cmssw/pull/19893#issuecomment-324371051

Comment (slava77) [reply tensorflow-sources.spec:15]: @davidlange6 @smuzaffar

**Pull request merged**